thelper/nn/resnet.py

```diff
@@ -262,7 +312,7 @@ def set_task(self, task):
             if self.fc is None or self.fc[len(self.fc) - 1].out_channels != num_classes:
                 if self.head_type == "fcn":
                     self.fc = torchvision.models.segmentation.fcn.FCNHead(self.out_features, num_classes)
-                else:
+                elif self.head_type == "deeplabv3":
                     self.fc = torchvision.models.segmentation.deeplabv3.DeepLabHead(self.out_features, num_classes)
```

Comment:
should we have an assert if None of the two?

Reply:
not necessarily --- a parent class might call that function, leave it default out to none, and implement its own layer after the call (pretty hacky, but on par for this messy file)